projects/baby_driver/scripts/test_adc_read.py

```diff
@@ -0,0 +1,49 @@
+"""This Module Tests methods in adc_read.py"""
+import unittest
+from unittest.mock import patch
+
+from adc_read import adc_read, NUM_PINS_PER_PORT
+from can_util import Message
+from gpio_port import GpioPort
+
+RAW_MAX = 4095
+CONVERTED_MAX = 3000
+
+class TestADCRead(unittest.TestCase):
+    """Test Babydriver's adc_read function"""
+
+    @patch('can_util.send_message')
+    @patch('can_util.next_message')
+    def test_check_range_pin(self):
+        """Tests Maximum values"""
+        self.assertLessEqual(adc_read(GpioPort.A, 0, True), RAW_MAX)
+        self.assertLessEqual(adc_read(GpioPort.A, 1, True), RAW_MAX)
+        self.assertLessEqual(adc_read(GpioPort.A, 2, True), RAW_MAX)
+        self.assertLessEqual(adc_read(GpioPort.A, 0, False), CONVERTED_MAX)
+        self.assertLessEqual(adc_read(GpioPort.A, 1, False), CONVERTED_MAX)
+        self.assertLessEqual(adc_read(GpioPort.A, 2, False), CONVERTED_MAX)
+
+    @patch('can_util.send_message')
+    @patch('can_util.next_message')
+    def test_fail_conditions(self, mock_send_message, mock_next_message):
```

Comment:
It turns out, per [the documentation](https://docs.python.org/3/library/unittest.mock-examples.html#patch-decorators), that the `patch` decorator actually passes the mock arguments in the reverse order that the decorators are laid out, so this argument list should be `self, mock_next_message, mock_send_message` to be correct.